src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleConnectionParameters.java

```diff
@@ -114,6 +136,90 @@ private String generateUrl() {
       buf.append("/");
       buf.append(serviceName);
     }
+
     return buf.toString();
   }
+
+  private String generateSecureUrl(StringBuilder buf) {
+    return new OracleTNSEntryBuilder()
+        .withProtocol("TCPS")
+        .withHost(host)
+        .withPort(port)
+        .withInstanceName(instance)
+        .withServiceName(serviceName)
+        .build(buf);
+  }
+
+  private void checkInstanceAndServiceName() {
+    if (instance != null && serviceName != null) {
+      String errorMessage = "Instance (SID) : [" + instance + "] and Service Name : [" + serviceName
+          + "] were provided at the same time, please configure only one";
+
+      throw new ModuleException(errorMessage, CONNECTIVITY, new ConnectionException(errorMessage));
+    }
+  }
+
+  /**
+   *
+   * Builder class to facilitate the creation of Oracle's URLs that use TNS Entries
+   *
+   */
+  public static final class OracleTNSEntryBuilder {
+
+    private String protocol;
+    private String host;
+    private Integer port;
+    private String instanceName;
+    private String serviceName;
+
+    private OracleTNSEntryBuilder() {}
+
+    public OracleTNSEntryBuilder withProtocol(String protocol) {
+      this.protocol = protocol;
+      return this;
+    }
+
+    public OracleTNSEntryBuilder withHost(String host) {
+      this.host = host;
+      return this;
+    }
+
+    public OracleTNSEntryBuilder withPort(Integer port) {
+      this.port = port;
+      return this;
+    }
+
+    public OracleTNSEntryBuilder withInstanceName(String instanceName) {
+      this.instanceName = instanceName;
+      return this;
+    }
+
+    public OracleTNSEntryBuilder withServiceName(String serviceName) {
+      this.serviceName = serviceName;
+      return this;
+    }
+
+    public String build(StringBuilder buf) {
```

Comment:
bad use of OOO. This receives a buffer of unknown state and content, mutates it and then returns the to string of that buf. That means that:

* you don't know who is invoking this method nor what's happening after, meaning the buffer will potentially receive `toString()` twice 
* You don't know the state of the buffer , which means you don't really know what is this class returning
* This class is operating under the unspoken assumption that this is the termination of the url building process.